### PR TITLE
bgp: Fix localas_ibgp inconsistencies and issues

### PR DESCRIPTION
**Pull request by jbemmel**
This PR fixes some corner case issues around ```localas_ibgp``` sessions:
* The ebgp.multihop plugin does not handle ```localas_ibgp``` sessions that might occur
* There is a misspelled key used to populate the ```bgp.community``` dictionary
* Users by default get settings that include ```localas_ibgp```, but cannot explicitly configure settings for this type of sessions

Fixes:
* #1663 
* #1670 

Note: The fact that the ```bgp.community``` key mismatch has gone undetected for 11 months implies a lack of coverage for CI/CD test cases, happy to add a test case if it is deemed important enough

The PR adds an entry to the ```integration/bgp.multihop/01-global``` test but no actual validation; it might be better to put this in a dedicated test case instead

### CODE REVIEW

File: netsim/extra/ebgp.multihop/plugin.py [hunk 1]
```diff
@@ -201,6 +201,13 @@ def fix_vrf_loopbacks(ndata: Box, topology: Box) -> None:
     else:
       ngb._source_ifname = ndata.loopback.ifname
 
+'''
+fix_localas_ibgp_neighbors - remove 'multihop' from any localas_ibgp neighbors
+'''
+def fix_localas_ibgp_neighbors(ndata: Box, topology: Box) -> None:
```
Comment (ipspace) [inline netsim/extra/ebgp.multihop/plugin.py:207]: I could not find where these changes are justified. Please remove them from this PR.

File: tests/integration/bgp.multihop/01-global.yml [hunk 2]
```diff
@@ -29,6 +29,10 @@ bgp.multihop.sessions:
 - dut:
   pe2:
     passive: True
+- dut:
```
Comment (ipspace) [inline tests/integration/bgp.multihop/01-global.yml:32]: This makes bgp.multihop functionality dependent on the bgp.local_as functionality. Remove from the PR.

File: netsim/modules/bgp.yml [hunk 3]
```diff
@@ -29,11 +29,11 @@ attributes:
...
     as_list: dict
     sessions:
-      ipv4: [ ibgp, ebgp ]
-      ipv6: [ ibgp, ebgp ]
+      ipv4: [ ibgp, ebgp, localas_ibgp ]
+      ipv6: [ ibgp, ebgp, localas_ibgp ]
     activate:
-      ipv4: [ ibgp, ebgp ]
-      ipv6: [ ibgp, ebgp ]
+      ipv4: [ ibgp, ebgp, localas_ibgp ]
```
Comment (ipspace) [inline netsim/modules/bgp.yml:35]: Most of the templates treat ibgp and localas_ibgp sessions the same way. What is the justification for them being different?

**Review by ipspace — CHANGES_REQUESTED** [review]
* Several files have to be removed from the PR.
* Most templates do not make the difference between ibgp and localas_ibgp for the AF activation purposes. We will not change the templates, but the behavior has to be documented. Please adjust the `bgp.md` if needed.

File: netsim/modules/bgp.yml (hunk 3 above)
Comment (jbemmel) [reply netsim/modules/bgp.yml:35]: We could decide to remove the ```localas_ibgp``` enum value and only support ```[ ibgp, ebgp ]```, but at the moment bgp neighbors of type ```localas_ibgp``` are being generated when ```bgp.local_as``` is set equal to the neighbor AS. The default ```activate``` and ```sessions``` values - if not set in a lab - include such 'localas_ibgp' sessions due to the Python string constant, but if I set the global sessions to some value there is no way to get such localas_ibgp sessions activated. They disappear silently

The justification is based on being able to configure all valid values, with consistency between the YAML files and what's defined by the Python source that accompanies them

File: netsim/extra/ebgp.multihop/plugin.py (hunk 1 above)
Comment (jbemmel) [reply netsim/extra/ebgp.multihop/plugin.py]: replaced by separate PR

File: tests/integration/bgp.multihop/01-global.yml (hunk 2 above)
Comment (jbemmel) [reply tests/integration/bgp.multihop/01-global.yml]: moved to separate test case in different PR

**Pull request merged**